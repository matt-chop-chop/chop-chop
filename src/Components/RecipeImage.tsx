import { Box, Flex, Image } from "@chakra-ui/react";

type RecipeImageProps = {
  image: string;
};

const RecipeImage = ({ image }: RecipeImageProps) => {
  return (
    <Box>
      {image.length !== 0 ? (
        <Image
          alt="A recipe image"
          maxHeight={["200px", "300px", "400px"]}
          objectFit="cover"
          src={image}
          width="100%"
        />
      ) : (
        <Flex
          alignItems="center"
          background="var(--chakra-colors-light-background)"
          height={["200px", "300px", "400px"]}
          justifyContent="center"
          objectFit="cover"
          p={1}
          width="100%"
        >
          <Image alt="Chop Chop Logo" height="75%" src="/images/logo.png" />
        </Flex>
      )}
    </Box>
  );
};

export default RecipeImage;
