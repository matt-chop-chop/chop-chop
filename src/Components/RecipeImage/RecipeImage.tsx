import { Flex, Image } from "@chakra-ui/react";
import React from "react";

export type RecipeImageProps = {
  image: string;
  maxHeight?: string[];
  defaultLogoWidth?: string[];
};

const RecipeImage = ({
  image,
  maxHeight = ["200px", "300px", "400px"],
  defaultLogoWidth = [],
}: RecipeImageProps) => {
  return (
    <Image
      alt="A recipe image"
      fallback={
        <Flex
          alignItems="center"
          background="var(--chakra-colors-light-background)"
          height={maxHeight}
          justifyContent="center"
          objectFit="cover"
          p={1}
          width="100%"
        >
          <Image
            alt="Chop Chop Logo"
            height="75%"
            src="/images/logo.png"
            width={defaultLogoWidth}
          />
        </Flex>
      }
      maxHeight={maxHeight}
      objectFit="cover"
      src={image}
      width="100%"
    />
  );
};

export default RecipeImage;
