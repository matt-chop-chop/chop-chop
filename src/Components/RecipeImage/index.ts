export { default as RecipeImage } from "./RecipeImage";
