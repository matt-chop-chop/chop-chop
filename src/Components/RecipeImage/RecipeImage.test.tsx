import { shallow } from "enzyme";
import React from "react";
import RecipeImage, { RecipeImageProps } from "./RecipeImage";

const props: RecipeImageProps = {
  image: "/test",
  maxHeight: ["100%"],
  defaultLogoWidth: ["100%"],
};

describe("RecipeImage", () => {
  it("is exported from RecipeImage", () => {
    expect(RecipeImage).toBeDefined();
  });

  it("renders as expected ", () => {
    const wrapper = shallow(<RecipeImage {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders with default props ", () => {
    const wrapper = shallow(<RecipeImage image={props.image} />);

    expect(wrapper).toMatchSnapshot();
  });
});
