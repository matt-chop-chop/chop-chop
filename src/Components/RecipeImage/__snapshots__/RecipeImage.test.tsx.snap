// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeImage renders as expected  1`] = `
<Image
  alt="A recipe image"
  fallback={
    <Flex
      alignItems="center"
      background="var(--chakra-colors-light-background)"
      height={
        [
          "100%",
        ]
      }
      justifyContent="center"
      objectFit="cover"
      p={1}
      width="100%"
    >
      <Image
        alt="Chop Chop Logo"
        height="75%"
        src="/images/logo.png"
        width={
          [
            "100%",
          ]
        }
      />
    </Flex>
  }
  maxHeight={
    [
      "100%",
    ]
  }
  objectFit="cover"
  src="/test"
  width="100%"
/>
`;

exports[`RecipeImage renders with default props  1`] = `
<Image
  alt="A recipe image"
  fallback={
    <Flex
      alignItems="center"
      background="var(--chakra-colors-light-background)"
      height={
        [
          "200px",
          "300px",
          "400px",
        ]
      }
      justifyContent="center"
      objectFit="cover"
      p={1}
      width="100%"
    >
      <Image
        alt="Chop Chop Logo"
        height="75%"
        src="/images/logo.png"
        width={[]}
      />
    </Flex>
  }
  maxHeight={
    [
      "200px",
      "300px",
      "400px",
    ]
  }
  objectFit="cover"
  src="/test"
  width="100%"
/>
`;
